src/caffe/util/im2col.cpp

```diff
@@ -10,9 +10,12 @@ void im2col_cpu(const Dtype* data_im, const int channels,
     const int height, const int width, const int kernel_h, const int kernel_w,
     const int pad_h, const int pad_w,
     const int stride_h, const int stride_w,
+    const int dilation_h, const int dilation_w,
     Dtype* data_col) {
-  const int height_col = (height + 2 * pad_h - kernel_h) / stride_h + 1;
-  const int width_col = (width + 2 * pad_w - kernel_w) / stride_w + 1;
+  const int height_col = (height + 2 * pad_h -
+      (dilation_h * (kernel_h - 1)  + 1)) / stride_h + 1;
+  const int width_col = (width + 2 * pad_w -
+      (dilation_w * (kernel_w - 1)  + 1)) / stride_w + 1;
```

Comment:
@fyu, there are two spaces before the first `+ 1` in each expression; in Caffe we always have single spaces around operators. (Perhaps this was intended as perceptual grouping, but it stands out as floating space to me, and we don't do that kind of grouping elsewhere in Caffe, and the precedence is quite clear here.)